### PR TITLE
Rewrite the documentation

### PR DESCRIPTION
**Pull request by mhauru**
I rewrote large parts of the READMEs and added much new stuff. It's not done done, this is just the first pass, but I think would be good to get comments and then get this merged, and we can keep on building on this together.

One question @EdwinB12  and I talked about that we need to decide is whether we should have multiple READMEs for the many subparts of the repo or one massive one with a lot of headings. I went with the former, but actually thought the latter might be better in the end, although Edwin then did some work trying to convince me otherwise.

### CODE REVIEW

File: dtbase/backend/README.md [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
-## Authentication
+The backend is a web app implemented using FastAPI. It takes in REST requests and returns responses.
 
-To be able to access any of the API end points you need an authentication token.
-You can get one from the `/auth/login` endpoint using a username and a password.
-Once you've obtained a token, you need to add it to header of any other API calls you make as a bearer token.
-So if `/auth/login` returned
+## Code structure
+* `run.sh`. This is script you call to run the FastAPI app.
+* `run_localdb.sh`. Just like `run.sh` except sources a different file of secrets and ensures that a local PostgreSQL server is running.
```
Comment (EdwinB12) [inline dtbase/backend/README.md:9]: Lets try and reduce this down to a single run.sh file when I merge the local installation.

File: dtbase/ingress/README.md [hunk 2]
```diff
@@ -114,9 +130,8 @@ weather_ingress = OpenWeatherDataIngress()
 weather_ingress.ingress_data(dt_from, dt_to)
```
Comment (EdwinB12) [inline dtbase/ingress/README.md:130]:
```suggestion
weather_ingress(dt_user_email=blahblah@email.com,
    dt_user_password="this is a very bad password", 
    dt_from=dt_from, 
    dt_to=dt_to)
```

File: dtbase/backend/README.md [hunk 3]
```diff
@@ -24,22 +39,26 @@ then you would call the other end points with the following in the header of the
...
 
-## Sensors
+### Sensors
 
 The Sensor data model is as follows.   Every _Sensor_ has a _SensorType_ which in turn specifies the variable(s) it can measure - these are known as _SensorMeasures_.  Each _SensorMeasure_ specifies its datatype (float, int, string, or bool), and these are used to define the type of the corresponding _SensorXYZReadings_.   A _Sensor_ may also have a _SensorLocation_, which specifies a _Location_ as defined above, and a time window (possibly open-ended) when the sensor was at that location.
 
-## Models
+### Models
 
 TODO: Write a summary of how storing model data works.
```
Comment (EdwinB12) [inline dtbase/backend/README.md:54]: TODO here that we should complete before merging for completeness

**Review by EdwinB12 — CHANGES_REQUESTED** [review]
Fabulous stuff. Some small changes and suggestions but generally fabulous! 

Now, don't hate me, but I want to reopen the discussion about putting this into a single docs readme. Think you originally argued this and having seen it laid out, I agree.

**Comment by EdwinB12** [comment]
Oh and lets add the simple ingress example in this PR.

File: README.md [hunk 4]
```diff
@@ -1,47 +1,73 @@
...
+Some things DTBase does not, currently, have:
+* Anything related to physical 3D structure of the objects that it is twinning. We have a very basic system for tracking locations in the twin, and for instance assigning sensors to locations, but this remains underdeveloped.
+* Comprehensive visualisation tools. The frontend has basic time series plotting capabilities, but that's it.
+* Sophisticated model orchestration. The services infrastructure can be used to store parameters for models and run them on demand, but the rest remains in development.
+* Anything for using the digital twin to control the real asset.
 
-We try to design the code base so as many of DTBase's features as possible can be deployed either locally or via Azure. There is no reason that DTBase couldn't be deployed on other cloud services, but Azure is what the developers have access to and therefore is the default service accomodated.
+DTBase may not be for you if
```
Comment (EdwinB12) [inline README.md:29]: Lets make headings like this a ### header

File: DeveloperDocs.md [hunk 5]
```diff
@@ -12,102 +11,101 @@ pip install '.[dev]'
...
-A tip: When developing the frontend, it can be very handy to run it with `FLASK_DEBUG=true DT_CONFIG_MODE=Auto-login ./run.sh`.
-The first environment variable makes it such that Flask restarts every time you make a change to the code, and the second one makes Flask automatically log in as the default user.
-This way when you make code changes, you can see the effect immediately in your browser without having to restart and/or log in.
+    `createdb --host localhost --username postgres dt_dev`
 
-### Running with an Azure Deployed Database
+If you've done this setup once and then e.g. rebooted your machine, all you should need to do is run `docker start dt_dev` to restart the existing Docker container.
 
-1. Copy the file `.secrets/dtenv_template.sh` to `.secrets/dtenv.sh` and populate this file with values for the various environment variables (ask an existing developer for these). These environment variables will point towards an already existing database hosted on Azure.
+### Running the tests
```
[the remark this answers is not in the document]
Comment (mhauru) [reply DeveloperDocs.md]: Done

File: dtbase/backend/README.md [hunk 6]
```diff
@@ -24,22 +39,26 @@ then you would call the other end points with the following in the header of the
...
 
-## Users
+### Users
 
 TODO: Write a summary of how user management works.
+
+## The Default User
+
+When starting a new deployment of a DTBase-based digital twin one encounters a chicken-and-egg dilemma: To be able to create users, or do anything really, with the backend, one needs to first have a registered user (the `/user/create-user` endpoint requires a valid JWT token like every other endpoint). To way out of this is the default user. If one sets the environment variable `DT_DEFAULT_USER_PASS` and starts the backend, at startup time a user with the "email" `default_user@localhost` is created, with the given password. One can use this to log in and create some proper users. One should then unset the `DT_DEFAULT_USER_PASS` environment variable and restart the backend. This causes the default user to be deleted.
```
[the remark this answers is not in the document]
Comment (mhauru) [reply dtbase/backend/README.md:62]: Done.

File: dtbase/ingress/README.md (hunk 2 above)
Comment (mhauru) [reply dtbase/ingress/README.md:130]: Done

File: README.md (hunk 4 above)
Comment (mhauru) [reply README.md:29]: Done

File: README.md [hunk 7]
```diff
@@ -1,47 +1,80 @@
 # DTBase
-A general base software package from which Digital Twins can be developed.
+A starting point from which digital twins can be developed.
 
-## Overview
+### What DTBase is
```
[the remark this answers is not in the document]
Comment (mhauru) [reply README.md:4]: I actually prefer the non-question version. Do you have a particular reason for preferring the (rhetorical) question one?

File: dtbase/backend/README.md [hunk 8]
```diff
@@ -24,22 +39,26 @@ then you would call the other end points with the following in the header of the
...
 
-## Models
+### Models
 
 TODO: Write a summary of how storing model data works.
 
-## Users
+### Users
 
 TODO: Write a summary of how user management works.
```
[the remark this answers is not in the document]
Comment (mhauru) [reply dtbase/backend/README.md:58]: I still expanded on this a bit more, now done

File: dtbase/backend/README.md (hunk 3 above)
Comment (mhauru) [reply dtbase/backend/README.md:54]: I added a couple of paragraphs here. There are no more TODO notes in the docs.

**Comment by mhauru** [comment]
@EdwinB12, I attended to all the points you raised (except for one, see above), and also did a read through of everything and made lots of small changes. Could you take another look please? I'm happy to merge once you are.

**Review by EdwinB12 — APPROVED** [review]

**Pull request merged**